src/sundials/sundials_profiler.c

```diff
@@ -496,22 +548,43 @@ int sunCompareTimes(const void* l, const void* r)
   double left_max;
   double right_max;
 
-  const SUNHashMapKeyValue left = *((SUNHashMapKeyValue*) l);
-  const SUNHashMapKeyValue right = *((SUNHashMapKeyValue*) r);
+  const SUNHashMapKeyValue left  = *((SUNHashMapKeyValue*)l);
+  const SUNHashMapKeyValue right = *((SUNHashMapKeyValue*)r);
+
+  if (left == NULL && right == NULL) return (0);
+  if (left == NULL) return (1);
+  if (right == NULL) return (-1);
 
-  if (left == NULL && right == NULL)
-    return(0);
-  if (left == NULL)
-    return(1);
-  if (right == NULL)
-    return(-1);
+  left_max  = ((sunTimerStruct*)left->value)->maximum;
+  right_max = ((sunTimerStruct*)right->value)->maximum;
 
-  left_max = ((sunTimerStruct*) left->value)->maximum;
-  right_max = ((sunTimerStruct*) right->value)->maximum;
+  if (left_max < right_max) return (1);
+  if (left_max > right_max) return (-1);
+  return (0);
+}
+
+int sunclock_gettime_monotonic(sunTimespec* ts)
+{
+#if (defined(WIN32) || defined(_WIN32)) && !defined(SUNDIALS_HAVE_POSIX_TIMERS)
+  static LARGE_INTEGER ticks_per_sec;
+  LARGE_INTEGER ticks;
 
-  if (left_max < right_max)
-    return(1);
-  if (left_max > right_max)
-    return(-1);
-  return(0);
+  if (!ticks_per_sec.QuadPart)
+  {
+    QueryPerformanceFrequency(&ticks_per_sec);
+    if (!ticks_per_sec.QuadPart) { return -1; }
+  }
+
+  QueryPerformanceCounter(&ticks);
+
+  /* QueryPerformanceCounter is ticks in microseconds */
+
+  ts->tv_sec  = (long)(ticks.QuadPart / ticks_per_sec.QuadPart);
+  ts->tv_nsec = (long)(((ticks.QuadPart % ticks_per_sec.QuadPart) * 1000000) /
+                       ticks_per_sec.QuadPart);
+
+  return 0;
+#else
+  return clock_gettime(CLOCK_MONOTONIC, ts);
+#endif
```

Comment:
Switch the order to simplify the `#if` to `#if defined(SUNDIALS_HAVE_POSIX_TIMERS)`

Reply:
I am not sure I follow. We would still have to check if it's WIN32, so it would only rearrange the logic rather than simplify.

Reply:
It would simplify the `#if` from
```
#if (defined(WIN32) || defined(_WIN32)) && !defined(SUNDIALS_HAVE_POSIX_TIMERS)
...
#else
...
#endif
```
to
```
#if defined(SUNDIALS_HAVE_POSIX_TIMERS)
...
#else
...
#endif
```